cmake/CMakeLists.txt

```diff
@@ -0,0 +1,31 @@
+# Minimum CMake required
+cmake_minimum_required(VERSION 3.1)
+
+# Project
+project(onnxir C CXX)
+
+# Set C++11 as standard for the whole project
+set(CMAKE_CXX_STANDARD 11)
+
+# External dependencies
+set(CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/external)
+set(ONNXIR_ROOT ${PROJECT_SOURCE_DIR}/..)
+if (WIN32)
+    # parallel build
+    add_compile_options(/MP)
+    add_compile_options(/WX)
+    set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /WX")
+    set(CMAKE_STATIC_LINKER_FLAGS "${CMAKE_STATIC_LINKER_FLAGS} /WX")
+endif()
+add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/../third_party/pybind11 ${CMAKE_CURRENT_BINARY_DIR}/pybind11)
```

Comment:
Since pybind11 is a header only library, we probably don't need a binary dir for add_subdirectory I assume?

Reply:
you're right. it's not needed to add a binary dir.

Because I'm now putting cmakelists.txt in cmake folder instead of root folder, it creates out of source tree case. In this case, cmake/add_subdirectory asks to put a binary dir. I personally don't want to move the cmakelists.txt into root folder btw.

Reply:
Resolved this now by moving the CmakeLists.txt file in root folder now.